File: src/Noticia.java
public class Noticia extends Publicacion {
  private String urlFuente;

  private String autor;

  // Constructors
  public Noticia() {}

  public Noticia(String titulo, String contenido, String imagenUrl, String urlFuente, String autor) {
    super(titulo, contenido, imagenUrl);
    this.urlFuente = urlFuente;
    this.autor = autor;
  }

  // Getters y Setters
  public String getUrlFuente() { return urlFuente; }
  public void setUrlFuente(String urlFuente) { this.urlFuente = urlFuente; }

  public String getAutor() { return autor; }
  public void setAutor(String autor) { this.autor = autor; }
}

File: app/model/entities/Noticia.ts
import { Publicacion } from "./Publicacion";

export class Noticia extends Publicacion {
  urlFuente?: string;
  autor?: string;

  constructor(titulo?: string, contenido?: string, imagenUrl?: string, urlFuente?: string, autor?: string) {
    super(titulo, contenido, imagenUrl);
    if (urlFuente && autor) {
      this.urlFuente = urlFuente;
      this.autor = autor;
    }
  }


}